src/document/json/rht_pq_map.rs

```diff
@@ -0,0 +1,272 @@
+use crate::document::json::element::Element;
+use crate::document::time::ticket::Ticket;
+
+use std::cell::RefCell;
+use std::cmp::Ordering;
+use std::collections::{BinaryHeap, HashMap};
+use std::rc::Rc;
+
+use thiserror::Error;
+
+type BoxedElement = Box<dyn Element>;
+
+#[derive(Debug, Error)]
+enum RHTPQMapError {
+    #[error("fail to find : {0}")]
+    ElementNotFound(String),
+}
+
+struct RHTPQMapNode {
+    key: String,
+    element: BoxedElement,
+}
```

Comment:
```suggestion
struct RHTPQMapNode<E: Element> {
    key: String,
    element: E,
}
```
How about using generic?
If using generic then not need to Box.

Reply:
I didn't know that. Thank you!